src/controllers/legacycontrollermapping.h

```diff
@@ -191,5 +266,7 @@ class LegacyControllerMapping {
     QString m_schemaVersion;
     QString m_mixxxVersion;
 
+    QList<std::shared_ptr<AbstractLegacyControllerSetting>> m_settings;
```

Comment:
Is the shared ownership correct/required?

Reply:
It isn't, however you cannot use a Qt container with a `unique_ptr`.

Reply:
Yes, that would be an improvement.

Reply:
Apologies, there is actually a shared ownership between the setting item and the appointing itself. Since setting mutation isn't possible once the mapping is started, we could however clone there instance. (when new setting gets applied, the mapping gets reloaded). What would you prefer?

Reply:
I think this got lost - the code is not changed. If I see it right, @daschuer suggested `std::vector` in the remark for src/controllers/legacycontrollermappingfilehandler.cpp.

Reply:
My understanding was that I should use a `std::vector` if I could replace the `std::shared_ptr` with a `std:: unique_ptr` but since I need to keep the shared pointer, I thought it might be best to keep the Qt container.

Reply:
@daschuer Do you aggree here?

Reply:
@daschuer Do you aggree with Antoine here?

Reply:
Yes.